### PR TITLE
refactor(workflow): centralize transition rules

### LINKED ISSUE #78: [Feature] 添加纯工作流状态转移模块
## Parent

#74

## What to build

Create a pure workflow transition module that owns command/event legality for the refactored job model. It should accept explicit domain inputs such as current phase, durable wait reason, execution state, chunk summary, and requested command, then return either the next state or a typed rejection reason.

The module must be usable by API command handling, restart recovery, runner/execution completion, and UI availability tests. Avoid fallback transitions and avoid forgiving unknown states; invalid states should fail loudly.

## Acceptance criteria

- [ ] The workflow module defines the approved commands/events for create validation complete, start/continue processing, user pause, execution stopped, process complete, process failed, retry failed chunks, merge, reset/delete, and restart recovery.
- [ ] Illegal transitions return explicit typed rejections or raise explicit domain errors rather than silently no-oping.
- [ ] API action handlers and UI action availability use this module or shared outputs from it instead of duplicating state rules.
- [ ] Table-driven tests cover valid and invalid transitions across every phase and wait reason.
- [ ] Existing ad hoc guard logic is removed or reduced to wrappers around the new module.

## Blocked by

#75

### PR DESCRIPTION
**Pull request by makoMakoGo**
## Summary
- Add an explicit pure workflow transition layer with `WorkflowContext`, typed command decisions, typed event transitions, and typed rejection codes.
- Route API command guards and snapshot `available_commands` through the shared workflow module.
- Use workflow events for runner completion, merge start/finish, and restart recovery state healing.
- Strengthen table-driven workflow tests and document the new coverage.

## Notes
- This PR is intentionally based on `main` and does not include the open #77 lifecycle PR changes.
- Invalid phase/state/wait combinations now fail explicitly instead of being normalized silently.

## Validation
- `uv run --frozen --no-sync ruff format --check`
- `uv run --frozen --no-sync ruff check`
- `uv run --frozen --no-sync python -m mypy novel_proofer`
- `uv run --frozen --no-sync python -m pytest -q -m "not llm_integration"`
- `uv run --frozen --no-sync python tools/export-openapi.py --check`
- `uv run --frozen --no-sync python tools/check-large-files.py`

Part of #74.
Closes #78.

## Summary by Sourcery

Centralize workflow transition logic into a typed workflow module and route API, runner, converters, and job store behavior through this shared transition layer.

New Features:
- Introduce a typed workflow context with explicit command decisions, event transitions, and rejection codes for job state management.
- Expose a workflow-driven computation of available job commands for use in API and UI responses.

Bug Fixes:
- Make invalid combinations of job state, phase, wait reason, and chunk status fail explicitly rather than being silently normalized.
- Ensure persisted jobs restored after restart transition to a valid paused state via the workflow event model.

Enhancements:
- Refactor runner, API endpoints, converters, and job store to use centralized workflow command and event helpers instead of duplicating state-transition rules.
- Add table-driven workflow tests covering command decisions, event transitions, wait-reason invariants, and available-commands derivation.
- Tighten job store update validation by reusing workflow state invariants in persistence logic.

Documentation:
- Document workflow test coverage and invariants in the test case catalog.

### CODE REVIEW

File: tests/runner/test_extra_coverage.py [hunk 1]
```diff
@@ -389,7 +389,13 @@ def test_retry_failed_and_resume_paused_branches(monkeypatch: pytest.MonkeyPatch
...
+            GLOBAL_JOBS.update(
+                job3_id,
+                phase="process",
+                work_dir=str(base / "w2"),
+                output_path=str(out_path),
+                cleanup_debug_dir=False,
+            )
             GLOBAL_JOBS.init_chunks(job3_id, total_chunks=1)
             GLOBAL_JOBS.update_chunk(job3_id, 0, state="done")
             runner.retry_failed_chunks(job3_id, llm)
```
[the remark this answers is not in the document]
Comment (makoMakoGo) [reply tests/runner/test_extra_coverage.py:401]: Reviewed. This path is already covered by tests/jobs/test_store.py::test_job_store_load_persisted_jobs_restores_running_job_as_paused: it persists a running process job, loads it through JobStore.load_persisted_jobs, and asserts paused/process/server_recovered plus processing chunk reset to pending. The implementation now routes restart restoration through WorkflowEvent.RESTART_RECOVERY, so adding a second runner test would duplicate the same boundary.

File: novel_proofer/api.py [hunk 2]
```diff
@@ -56,12 +56,35 @@
...
 logger = logging.getLogger(__name__)
 
 
+def _workflow_context_for_job(st: JobStatus) -> WorkflowContext:
+    return WorkflowContext.from_values(
+        state=st.state,
+        phase=st.phase,
+        wait_reason=st.wait_reason,
+        chunks=[chunk.state for chunk in st.chunk_statuses],
+    )
```
[the remark this answers is not in the document]
Comment (makoMakoGo) [reply novel_proofer/api.py]: Accepted. API workflow context construction now uses workflow_context_for_job, which reads JobStatus.chunk_counts through WorkflowContext.from_counts when counts are present, and only reads chunk_statuses for statuses that do not carry aggregate counts.

File: novel_proofer/jobs.py [hunk 3]
```diff
@@ -607,8 +614,19 @@ def _restore_loaded_job_after_restart(self, st: JobStatus) -> tuple[JobStatus, b
...
-            st.wait_reason = WaitReason.SERVER_RECOVERED
+            transition = require_event(
+                WorkflowContext.from_values(
+                    state=st.state,
+                    phase=st.phase,
+                    wait_reason=st.wait_reason,
+                    chunks=[chunk.state for chunk in st.chunk_statuses],
+                ),
+                WorkflowEvent.RESTART_RECOVERY,
+            )
```
[the remark this answers is not in the document]
Comment (makoMakoGo) [reply novel_proofer/jobs.py]: Accepted. Restart recovery now builds WorkflowContext.from_counts from JobStatus.chunk_counts when aggregate counts are present before applying WorkflowEvent.RESTART_RECOVERY.
